tests/unit/adapters/application-test.js

```diff
@@ -29,103 +29,187 @@ moduleFor('adapter:application', 'Unit | Adapter | application', {
   let promise = adapter.findQuery({ query: { sort:'-desc' } });
+
   assert.ok(typeof promise.then === 'function', 'returns a thenable');
-  assert.ok(adapter.fetch.calledOnce, 'fetch called');
-  assert.ok(adapter.fetch.calledWith('/posts?sort=-desc', { method: 'GET' }), 'fetch called with url?query and method:GET');
+  promise.then(() => {
+    assert.ok(adapter.fetch.calledOnce, 'fetch called');
+    assert.ok(
+      adapter.fetch.calledWith('/posts?sort=-desc', {method: 'GET'}),
+      'fetch called with url?query and method:GET'
+    );
+    done();
+  });
 });
 
 test('#findRelated', function(assert) {
+  assert.expect(3);
+  const done = assert.async();
+
+  // Resource to findRelated on.
+  let resource   = this.container.lookup('model:post').create(postMock.data);
+  let relatedUrl = resource.get('relationships.author.links.related');
+  const adapter  = this.subject({type: 'posts', url: '/posts'});
+
+  // We expect the authors service to be used.
   this.registry.register('service:authors', Adapter.extend({type: 'authors', url: '/authors'}));
-  let resource = this.container.lookup('model:post').create(postMock.data);
-  let url = resource.get( ['relationships', 'author', 'links', 'related'].join('.') );
-  const adapter = this.subject({type: 'posts', url: '/posts'});
   let service = this.container.lookup('service:authors');
-  let related = this.container.lookup('model:author').create(authorMock.data);
   sandbox.stub(service, 'fetch', function () { return RSVP.Promise.resolve(related); });
-  let promise = adapter.findRelated('author', url);
+
+  // The related resource we expect to find.
+  let related = this.container.lookup('model:author').create(authorMock.data);
+
+  let promise = adapter.findRelated('author', relatedUrl);
+
   assert.ok(typeof promise.then === 'function', 'returns a thenable');
-  assert.ok(service.fetch.calledOnce, 'authors service#fetch method called');
-  let expectURL = 'http://api.pixelhandler.com/api/v1/posts/1/author';
-  assert.ok(service.fetch.calledWith(expectURL, { method: 'GET' }), 'url for relation passed to service#fetch');
+  promise.then(() => {
+    assert.ok(service.fetch.calledOnce, 'authors service#fetch method called');
+    assert.ok(
+      service.fetch.calledWith(relatedUrl, {method: 'GET'}),
+      'url for relation passed to service#fetch'
+    );
+    done();
+  });
 });
 
 test('#findRelated can be called with optional type for the resource', function (assert) {
   assert.expect(4);
   const done = assert.async();
-  let PersonAdapter = Adapter.extend({type: 'people', url: '/people'});
-  PersonAdapter.reopenClass({ isServiceFactory: true });
-  this.registry.register('service:people', PersonAdapter.extend());
-  let service = this.container.lookup('service:people');
-  let supervisor = this.container.lookup('model:employee').create({
+
+  let EmployeeAdapter = Adapter.extend({type: 'employees', url: '/employees'});
```

Comment:
@aars the test was not using `polymorphism` I do have an acceptance test for that though.